tailcfg/tailcfg.go

```diff
@@ -454,6 +459,120 @@ type Hostinfo struct {
 	//       require changes to Hostinfo.Equal.
 }
 
+// View returns a read-only accessor for the Hostinfo object.
+func (hi *Hostinfo) View() HostinfoView { return HostinfoView{hi} }
+
+// HostinfoView is a read-only accessor for Hostinfo.
+// See Hostinfo.
+type HostinfoView struct {
+	// ж represents the underlying value.
+	// It is considered a lower-case letter so it is not exported.
+	ж *Hostinfo
+}
+
+func (v HostinfoView) MarshalJSON() ([]byte, error) {
+	return json.Marshal(v.ж)
+}
+
+func (v *HostinfoView) UnmarshalJSON(b []byte) error {
+	if v.ж != nil {
+		return errors.New("HostinfoView is already initialized")
+	}
+	if len(b) == 0 {
+		return nil
+	}
+	hi := &Hostinfo{}
+	if err := json.Unmarshal(b, hi); err != nil {
+		return err
+	}
+	v.ж = hi
+	return nil
+}
+
+// Valid reports whether the underlying struct is not nil.
+func (v HostinfoView) Valid() bool { return v.ж != nil }
+
+// AsStruct returns a deep-copy of the underlying struct.
+func (v HostinfoView) AsStruct() *Hostinfo { return v.ж.Clone() }
+
+// Clone creates a shallow-copy of the HostinfoView.
+// This is safe as the underlying struct is considered immutable.
+func (v *HostinfoView) Clone() *HostinfoView { return &HostinfoView{v.ж} }
```

Comment:
wait, why? I'm lost.

And why ever a pointer to a view?

Reply:
Oh this was me fighting with cloner. How do I make it play nice with views?

Reply:
Fixed cloner